src/batch/src/executor/join/mod.rs

```diff
@@ -70,6 +70,7 @@ impl JoinType {
             JoinTypeProst::RightSemi => JoinType::RightSemi,
             JoinTypeProst::RightAnti => JoinType::RightAnti,
             JoinTypeProst::FullOuter => JoinType::FullOuter,
+            _ => panic!("Unspecified"),
```

Comment:
We should avoid matching `_` in case we add a new type but forget to add it as an arm of `match`es.
Please use `Foo::Unspecified => unreachable!()`.